### PR TITLE
:heavy_plus_sign: Add global usage of `PoolIdLibrary` for `PoolKey` struct

### PR DESCRIPTION
**Pull request by GianfrancoBazzani**
This addition will allow the usage of the method `PoolKey.toId()` without require to import the `PoolIdLibrary` and use the `using PoolIdLibrary for PoolKey` directive.

### CODE REVIEW

File: src/types/PoolKey.sol [hunk 1]
```diff
@@ -3,6 +3,9 @@ pragma solidity ^0.8.0;
 
 import {Currency} from "./Currency.sol";
 import {IHooks} from "../interfaces/IHooks.sol";
+import {PoolIdLibrary} from "./PoolId.sol";
+
+using PoolIdLibrary for PoolKey global;
```
Comment (hensha256) [inline src/types/PoolKey.sol:8]: can you then remove `using PoolIdLibrary for PoolKey` frm all the contracts where it is used?

**Review by hensha256 — CHANGES_REQUESTED** [review]

File: src/types/PoolKey.sol (hunk 1 above)
Comment (hensha256) [reply src/types/PoolKey.sol:8]: Please can you also sign your commit :)

**Comment by GianfrancoBazzani** [comment]
* hey @hensha256 thanks for the suggestion!  It should be ok now. Added also corresponding PR to v4-periphery repo: https://github.com/Uniswap/v4-periphery/pull/300

**Review by hensha256 — DISMISSED** [review]

**Comment by hensha256** [comment]
Hi! Sorry you have some merge conflicts form another PR we merged. Please can you fix those?

**Comment by GianfrancoBazzani** [comment]
Done!

**Review by hensha256 — APPROVED** [review]

**Pull request merged**